### ISSUE TITLE
Folder creation?

### ISSUE THREAD

**Issue by avpavp** [safrazik/vue-file-agent#65]
Is there a way to create folders?  Or am I missing something?

Thanks.

**Comment by safrazik** [comment]
No, this is a UI component for web where folder or file cannot be created on the local file system.

But it would be a good feature to support folder icons along with other file types for preloaded fileRecords - where this component will be used in a web file manager (maybe that’s what you’re trying to achieve).

**Comment by drascom** [comment]
> > Is there a way to create folders? Or am I missing something?

in template add
`	<button :disabled="!fileRecordsForUpload.length" @click="uploadFiles(customerId)">
			Upload {{ fileRecordsForUpload.length }} files
		</button>`

in script add

`	uploadFiles: function(id) {
			this.$refs.vueFileAgent
				.upload(
					this.uploadUrl,
					this.uploadHeaders,
					this.fileRecordsForUpload,
					function createFormData(fileData) {
						var formData = new FormData();
						formData.append("id", id);
						formData.append("file", fileData.file); // this is important! - the actual file for upload
						return formData;
					}
				)}`

in server side php change upload function like below

` public function handleUpload($uploadedFile,$folder){
        // $filename = md5(microtime().$uploadedFile['name'].$uploadedFile['tmp_name']); // create a unique name
        $filename = $uploadedFile['name']; // or use the file name from client
        **if($folder != ''){
            $this->uploadDir=$this->uploadDir.'\\'.$folder;
            if (!file_exists($this->uploadDir)) {
                mkdir($this->uploadDir, 0777, true);
            }
        }**
        move_uploaded_file($uploadedFile['tmp_name'], $this->uploadDir.DIRECTORY_SEPARATOR.$filename);
        return ['my_key' => $filename]; // you can send any arbitrary data to client which will be saved in fileRecord.upload key in client, and will be sent back to server at update/delete request
    }`

**Comment by safrazik** [comment]
@drascom I'm sorry I don't understand your question. Can you format your question in a way that can be easily understood.

**Comment by drascom** [comment]
hi, it was an answer to "avpavp" to show create sub folder with posted data.
i know it is looking ugly but i realy cant figure out how to use backtick to format code sections

**Comment by drascom** [comment]
@safrazik it was not question just try to show the way creating sub folder with posted data while uploading file to @avpavp 
great script by the way thank you...
